rainier-core/src/main/scala/com/stripe/rainier/core/Categorical.scala

```diff
@@ -99,10 +99,27 @@ final case class Binomial(p: Real, k: Int) extends Distribution[Int] {
   val multi: Multinomial[Boolean] =
     Categorical.boolean(p).toMultinomial(k)
 
-  def generator: Generator[Int] =
-    multi.generator.map { m =>
+  def generator: Generator[Int] = {
+    val poissonGenerator = Poisson(p * k).generator.map { _.min(k) }
+    val normalGenerator = Normal(k * p, k * p * (1 - p)).generator.map {
+      _.toInt.max(0).min(k)
+    }
+    val binomialGenerator = multi.generator.map { m =>
       m.getOrElse(true, 0)
     }
+    Generator.from {
+      case (r, n) =>
+        val pDouble = n.toDouble(p)
+        val oneMinusP = n.toDouble(1 - p)
```

Comment:
nit: more efficient to write `1.0 - pDouble`

Reply:
oh, nice